tools/mo/openvino/tools/mo/main.py

```diff
@@ -308,25 +322,53 @@ def raise_ie_not_found():
         from openvino.tools.mo.front.onnx.register_custom_ops import get_front_classes
         import_extensions.load_dirs(argv.framework, extensions, get_front_classes)
 
-    return argv
 
+def check_fallback(argv : argparse.Namespace):
+    fallback_reasons = {}
 
-def prepare_ir(argv):
-    argv = arguments_post_parsing(argv)
+    if not any(deduce_framework_by_namespace(argv)): # no legacy path
+        return fallback_reasons
+
+    if argv.use_new_frontend or argv.use_legacy_frontend: # frontend chosen explicitly
```

Comment:
I think it is sufficient to have here just `if argv.use_new_frontend` because `moc_front_end` is None in case `use_legacy_frontend`.
Also please add a comment that
```
There is no possibility for fallback if an user strictly wants to use new frontend
```